presto-main/src/main/java/com/facebook/presto/operator/scalar/ArraysOverlapFunction.java

```diff
@@ -13,202 +13,65 @@
-            @SqlType("array(integer)") Block rightArray)
-    {
-        return genericArraysOverlap(leftArray, rightArray, INTEGER);
-    }
-
     @SqlNullable
-    @SqlType(StandardTypes.BOOLEAN)
-    public Boolean arraysOverlapBigInt(
-            @OperatorDependency(operator = LESS_THAN, argumentTypes = {"bigint", "bigint"}) MethodHandle lessThanFunction,
-            @SqlType("array(bigint)") Block leftArray,
-            @SqlType("array(bigint)") Block rightArray)
-    {
-        return genericArraysOverlap(leftArray, rightArray, BIGINT);
-    }
-
-    @SqlNullable
-    @TypeParameter("E")
-    @SqlType(StandardTypes.BOOLEAN)
-    public Boolean arraysOverlap(
-            @OperatorDependency(operator = LESS_THAN, argumentTypes = {"E", "E"}) MethodHandle lessThanFunction,
-            @TypeParameter("E") Type type,
-            @SqlType("array(E)") Block leftArray,
-            @SqlType("array(E)") Block rightArray)
+    public static Boolean arraysOverlap(
+            @TypeParameter("T") Type type,
+            @SqlType("array(T)") Block leftArray,
+            @SqlType("array(T)") Block rightArray)
     {
         int leftPositionCount = leftArray.getPositionCount();
         int rightPositionCount = rightArray.getPositionCount();
 
         if (leftPositionCount == 0 || rightPositionCount == 0) {
             return false;
         }
-
-        if (leftPositions == null || leftPositions.length < leftPositionCount) {
-            leftPositions = new int[leftPositionCount * 2];
-        }
-
-        if (rightPositions == null || rightPositions.length < rightPositionCount) {
-            rightPositions = new int[rightPositionCount * 2];
-        }
-
-        for (int i = 0; i < leftPositionCount; i++) {
-            leftPositions[i] = i;
-        }
-        for (int i = 0; i < rightPositionCount; i++) {
-            rightPositions[i] = i;
+        Block lookArray = leftArray;
+        Block itrArray = rightArray;
+        if (leftPositionCount > rightPositionCount) {
+            lookArray = rightArray;
+            itrArray = leftArray;
         }
-        IntArrays.quickSort(leftPositions, 0, leftPositionCount, intBlockCompare(type, leftArray));
-        IntArrays.quickSort(rightPositions, 0, rightPositionCount, intBlockCompare(type, rightArray));
 
-        int leftCurrentPosition = 0;
-        int rightCurrentPosition = 0;
-        while (leftCurrentPosition < leftPositionCount && rightCurrentPosition < rightPositionCount) {
-            if (leftArray.isNull(leftPositions[leftCurrentPosition]) || rightArray.isNull(rightPositions[rightCurrentPosition])) {
-                // Nulls are in the end of the array. Non-null elements do not overlap.
-                return null;
+        boolean hasNull = false;
+        TypedSet typedSet = new TypedSet(type, lookArray.getPositionCount(), "arraysOverlap");
+        for (int i = 0; i < lookArray.getPositionCount(); i++) {
+            if (lookArray.isNull(i)) {
```

Comment:
You don't need to do this

Reply:
Hi, are you referring to `if (lookArray.isNull(i)) {`, then for the case arrays_overlap(Array [NULL], Array [1, 2]), we need to return null as result right?

Reply:
So yeah see my next comment use addNoNull so that we can avoid double checking.